plugins/org.yakindu.sct.simulation.core.sexec/src/org/yakindu/sct/simulation/core/sexec/interpreter/StextStatementInterpreter.xtend

```diff
@@ -156,12 +158,14 @@ class StextStatementInterpreter extends AbstractStatementInterpreter {
 
 	def dispatch Object execute(EventRaisingExpression eventRaising) {
 		var event = context.resolve(eventRaising.event)
-		if (eventRaising.value != null) {
-			event.value = eventRaising.value.execute
-		}
+
 		if (event instanceof ExecutionEvent) {
-			(event as ExecutionEvent).raised = true
+			
+			val value = eventRaising.value?.execute	
+		
+			if (eventRaiser != null) event.raise(value) 
```

Comment:
I don't see eventRaiser variable used somewhere but for the null-check only, is that on purpose?